src/url.js

```diff
@@ -43,6 +43,33 @@ export function parseUrl(url) {
   return info;
 }
 
+/**
+ * Appends a query string field and value to a url. `key` and `value`
+ * will be ran through `encodeURIComponent` before appending.
+ * @param {string} url
+ * @param {string} key
+ * @param {string} value
+ * @return {string}
+ */
+export function addParamToUrl(url, key, value) {
+  const urlObj = parseUrl(url);
```

Comment:
Do you think it might be too heavy to create DOM elements for this? Maybe simple string scan for `?` and `#` would be better?